training-export-service/export.py

```
@@ -211,6 +211,36 @@ def stratified_split(
+    split. When that split is test (e.g. Nikitopoulos), the model never trains on
+    the style and scores ~0 on it however well the data is annotated (observed in
+    v6: val formula 0.27 but test formula 0.004). Splitting by PAGE puts ~80/10/10
+    of EVERY document's pages into train/val/test, so every publisher/formula
+    style is seen in training AND evaluated in val+test.
+
+    Why hashed (not random or index-based): the split must be PINNED — the same
+    (doc, page) must map to the same split on every run so re-annotation never
+    shuffles pages between splits and successive model versions stay directly
+    comparable. Python's built-in hash() is salted per-process (PYTHONHASHSEED),
+    so we use md5 for a stable, reproducible bucket."""
+    h = hashlib.md5(f"{doc_id}:{page_num}".encode('utf-8')).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)